File: obfuscated/src/com/genir/starfarer/combat/ai/system/SystemAI.java
package com.genir.starfarer.combat.ai.system;

import com.genir.starfarer.combat.entities.Ship;
import org.lwjgl.util.vector.Vector2f;

// OBFUSCATED
public interface SystemAI {
    // OBFUSCATED
    void systemAI_advance(float dt, Vector2f missileDangerDir, Vector2f collisionDangerDir, Ship target);
}

File: src/com/genir/aitweaks/core/features/shipai/systems/SystemAI.kt
package com.genir.aitweaks.core.features.shipai.systems

import com.fs.starfarer.api.combat.ShipAPI
import com.fs.starfarer.api.combat.ShipSystemAPI
import com.genir.aitweaks.core.features.shipai.CustomShipAI
import org.lwjgl.util.vector.Vector2f

abstract class SystemAI(val ai: CustomShipAI) {
    protected val ship: ShipAPI = ai.ship
    protected val system: ShipSystemAPI = ship.system

    abstract fun advance(dt: Float)

    open fun holdTargets(): Boolean = system.isOn

    open fun overrideHeading(): Vector2f? = null

    open fun overrideFacing(): Float? = null
}